File: go/fixture/go-mini/internal/report/wire.go
package report

import "io"

// Announce writes the catalog event for name to w, if the catalog knows the
// device.
func Announce(w io.Writer, catalog *Catalog, name string) {
	sink := NewSink(w)
	rep := NewReporter(sink, nil, nil)
	if d := catalog.Find(name); d != nil {
		rep.Record(d.Event())
	}
}

File: py/fixture/py-mini/internal/report/wire.py
"""Wiring of the reporter for one-off announcements."""

from typing import TextIO

from internal.report.catalog import Catalog
from internal.report.reporter import Reporter, Sink


def announce(w: TextIO, catalog_text: str, name: str) -> None:
    """Write the catalog event for name to w, if the catalog text lists the device."""
    sink = Sink(w)
    rep = Reporter(sink, None, None)
    d = Catalog.parse(catalog_text).find(name)
    if d is not None:
        rep.record(d.event())
